File: cse465-ComparativeProgramming/Homework 4 (Python C# mailmerge multiset)/mm.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSE465
{
    class mm
    {
        public const char DELIM = '\t';
        public static string merge(String tmpl, string[] fields, string[] replacements)
        {
            for (int i = 0; i < fields.Length; i++) {
                tmpl = tmpl.Replace("<<" + fields[i] + ">>", replacements[i]);
            }
            return tmpl;
        }
        public static void save(string filename, string text)
        {
            StreamWriter output = new StreamWriter(filename + ".txt");
            output.Write(text);
            output.Close();
        }
        static void Main(string[] args)
        {
            // Load the template
            StreamReader input = new StreamReader(args[args.Length-1]);
            string tmpl = input.ReadToEnd();
            input.Close();

            // Load the data file
            input = new StreamReader(args[args.Length-2]);

            //Get the fields
            string[] fields = input.ReadLine().Split(DELIM);

            string line, temp;
            string[] data;
            while((line = input.ReadLine()) != null)
            {
                data = line.Split(DELIM);
                temp = merge(tmpl, fields, data);
                save(data[1], temp);
            }
        }
    }
}

File: cse465-ComparativeProgramming/Term project/mm/mm.py
#!/usr/bin/python3

import sys

delim = '\t' # Specify the data delimiting character
dataFilename = sys.argv[1]
tmplFilename = sys.argv[2]

# Replace everything inside << >> with corresponding data
def merge(template, data):
	for field in data:
		template = template.replace('<<' + field + '>>', data[field])
	return template

# Save text to a file
def save(text, filename):
	with open(filename, 'w') as file:
		file.write(text)

# Build template string from multiple lines
tmpl = ""
with open(tmplFilename, 'r') as tmplFile:
	tmpl = "".join(tmplFile.readlines()) # Just because it's Python--why not

with open(dataFilename, 'r') as dataFile:
	# Get field names
	fields = dataFile.readline().rstrip().split(delim)
	# Go through each data line
	for line in dataFile:
		# Remove new line character -> get each string
		rpl_strings = line.rstrip().split(delim)
		# Create a dictionary out of field names and data
		replacements = dict(zip(fields, rpl_strings))
		# Merge dictionary with template
		text = merge(tmpl, replacements)
		# Save results
		save(text, replacements['ID'] + '.txt')
		